llvm/lib/CodeGen/MachineLICM.cpp

```diff
@@ -1731,3 +1769,20 @@ bool MachineLICMBase::isTgtHotterThanSrc(MachineBasicBlock *SrcBlock,
   // Compare the block frequency ratio with the threshold
   return Ratio > BlockFrequencyRatioThreshold;
 }
+
+template class llvm::MachineLICMBasePass<EarlyMachineLICMPass, true>;
+template class llvm::MachineLICMBasePass<MachineLICMPass, false>;
```

Comment:
Move them after the definition, see https://en.cppreference.com/w/cpp/language/class_template#Class_template_instantiation